.github/workflows/build-and-push-docker-image.yml

```
@@ -0,0 +1,53 @@
+---
+
+name: 🏗️
+
+on:   # yamllint disable-line rule:truthy
+  pull_request:
+  push:
+    branches: ["release/*", "unstable/*"]
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: Docker image tag (optional, defaults to Git ref)
+        required: false
+        type: string
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+    - uses: actions/checkout@v4
+    - name: Build Docker image
+      run: |
+        DOCKER_TAG="${DOCKER_TAG/'/'/'-'}"
+        DOCKER_TAG_MAJOR=$(echo "$DOCKER_TAG" | cut -d '.' -f 1)
+        DOCKER_TAG_MAJOR_MINOR=$(echo "$DOCKER_TAG" | cut -d '.' -f 1-2)
+        IMAGE="ghcr.io/$GITHUB_REPOSITORY:${DOCKER_TAG}"
+        IMAGE_MAJOR="ghcr.io/$GITHUB_REPOSITORY:${DOCKER_TAG_MAJOR}"
+        IMAGE_MAJOR_MINOR="ghcr.io/$GITHUB_REPOSITORY:${DOCKER_TAG_MAJOR_MINOR}"
+        echo "IMAGE=$IMAGE" >>"$GITHUB_ENV"
+        echo "IMAGE_MAJOR=$IMAGE_MAJOR" >>"$GITHUB_ENV"
+        echo "IMAGE_MAJOR_MINOR=$IMAGE_MAJOR_MINOR" >>"$GITHUB_ENV"
+        docker build . \
+          --build-arg BUILDKIT_INLINE_CACHE=1 \
+          --cache-from $IMAGE \
+          --tag $IMAGE
+        docker tag $IMAGE $IMAGE_MAJOR
+        docker tag $IMAGE $IMAGE_MAJOR_MINOR
+      env:
+        DOCKER_TAG: ${{ inputs.tag || github.ref_name }}
+    - name: Log in to GHCR
+      if: github.event_name != 'pull_request'
+      run: >-
+        echo ${{ secrets.GITHUB_TOKEN }} |
+          docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
+    - name: Push Docker image to GHCR
+      if: github.event_name != 'pull_request'
+      run: |
+        docker push $IMAGE
+        docker push $IMAGE_MAJOR
+        docker push $IMAGE_MAJOR_MINOR
+  smoke-test:
```

There should be a dependency on the previous job
```suggestion
  smoke-test:
    needs:
    - build-and-push
```